### DISCUSSION TITLE
export dependency libraries automatically

### DISCUSSION THREAD

**Discussion by ClausKlein** [friendlyanon/cmake-init#52] [Ideas]
I had to add the known dependency for a library manually to [cmake/install-config.cmake](https://github.com/ClausKlein/asio/blob/dc663360dd3a1bd4af82b9497f96eb226b8febc5/asio/cmake/install-config.cmake)

But it should and may be generated.

```cmake
# ---- add dependenc libraries ----

 find_package(Threads REQUIRED)
 target_link_libraries(asio_asio INTERFACE Threads::Threads)
```

see https://discourse.cmake.org/t/featuresummary-module-how-to-get-all-dependency-libraries-of-a-target-and-their-versions/5308

**Comment by friendlyanon** [comment]
That gets generated when a package manager is used for the project types where it's appropriate: https://github.com/friendlyanon/cmake-init/blob/develop/cmake-init/templates/common/cmake/install-config.cmake

There is also the transitive dependency example from before package manager support was added: https://github.com/friendlyanon/cmake-init-transitive
